library/std/src/sys_common/thread_info.rs

```diff
@@ -30,6 +30,15 @@ impl ThreadInfo {
     }
 }
 
+/// Get an address that is unique per running thread.
+///
+/// This can be used as a non-null usize-sized ID.
+pub fn current_thread_unique_ptr() -> usize {
+    // Use a non-drop type to make sure it's still available during thread destruction.
+    thread_local! { static X: u8 = 0 }
+    X.with(|x| <*const _>::addr(x))
+}
```

Comment:
Can this be in library/std/src/sys/unix/locks/futex.rs instead?

Reply:
Oh yes, I'll move it. It was here because a previous version took the address of THREAD_INFO.